File: 134-gas-station/gas-station.ts
function canCompleteCircuit(gas: number[], cost: number[]): number {
    const sum = (arr : number[]) => arr.reduce((num,acc)=>acc + num,0);
    
    if(sum(gas) < sum(cost)) return -1;
    let total = 0;
    let result = 0;
    for(let i = 0; i < gas.length; i ++){
        total += (gas[i] - cost[i]);

        if (total < 0){
            total = 0;
            result = i + 1;
        }
    }
    return result;
};

File: 134-gas-station/gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int totalGas = 0;
        int totalCost = 0;
        for(int i = 0; i < gas.length; i ++){
            totalGas += gas[i];
            totalCost += cost[i];
        }
        if(totalGas < totalCost) return -1;

        int tank = 0;
        int start = 0;
        for(int i = 0; i < gas.length; i ++) {
            tank += gas[i] - cost[i];
            if(tank < 0) {
                tank = 0;
                System.out.printf("i : %d, tank : %d \n", i, tank);
                start = i + 1;
            }
        }
        return start;
    }
}